app/lib/dartdoc/dartdoc_runner.dart

```diff
@@ -19,7 +28,68 @@ class DartdocRunner implements TaskRunner {
 
   @override
   Future<bool> runTask(Task task) async {
-    // TODO: implement doc generation and upload
-    return false; // no race detected
+    final tempDir =
+        await Directory.systemTemp.createTemp('pub-dartlang-dartdoc');
+    final pubCacheDir = await tempDir.resolveSymbolicLinks();
```

Comment:
Ideally we would reuse a pub cache, instead of just re-downloading all transitive dependencies for every single package we generate documentation for.

What is this `PubEnvironment` stuff doing?

Is `pubCacheDir` the real `PUB_CACHE` or is it a directory where this particular (package, version) tarball will be downloaded to?

Reply:
We have had an issue with the `analyzer` service early on, when I've left out the temp dir creation and cleanup: the packages fill up the disk quickly.

In this case `pubCacheDir` is a clean environment, initialized and clear for every `dartdoc` run. However, `pub cache` *does not* download the transitive dependencies, only the specified package, so it is fine to use it a a clean env.

`PubEnvironment` encapsulates the various executables, commands and environment variables for `pana`. It felt better to reuse it here, than to reinvent everything.